snuba/datasets/errors_replacer.py

```diff
@@ -172,7 +172,9 @@ def process_message(self, message: ReplacementMessage) -> Optional[Replacement]:
                 self.__schema,
             )
         elif type_ == "tombstone_events":
-            processed = process_tombstone_events(event, self.__required_columns)
+            processed = process_tombstone_events(
+                event, self.__required_columns, self.__state_name
+            )
```

Comment:
Just a clarification, we use this only for reprocessing right? This means that as soon as the events table is removed we will always have to add a tombstone for replacement and this logic based on state name will not be needed anymore. Is this correct ?

Reply:
That's all correct. However, we will only issue tombstones for replacements that change the primary hash of the event. If the primary hash stays the same there's no tombstone required.